tests/lib/prepare-restore.sh

```diff
@@ -117,7 +118,7 @@ build_rpm() {
     cp -ra -- * "/tmp/pkg/snapd-$version/"
     mkdir -p "$rpm_dir/SOURCES"
     # shellcheck disable=SC2086
-    (cd /tmp/pkg && tar "-c${archive_compression}f" "$rpm_dir/SOURCES/$archive_name" $extra_tar_args "snapd-$version")
+    (cd /tmp/pkg && tar "-c${archive_compression}f" "$rpm_dir/SOURCES/$archive_name" "${extra_tar_args[@]}" "snapd-$version")
```

Comment:
I believe this fails in practice:
```
+ tar -cJf /usr/src/packages/SOURCES/snapd_1337.2.34~pre1.vendor.tar.xz '' snapd-1337.2.34~pre1
tar: Substituting `.' for empty member name
tar: : Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
```

Reply:
Heh, shell quoting and preserving single quotes is crazy.